src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/ZeroProtocolHandlerBase.cs

```diff
@@ -58,6 +58,7 @@ CancellationToken token
             Task firstTask = await Task.WhenAny(task, Task.Delay(Timeouts.Eth, compositeCancellation.Token));
             if (firstTask.IsCanceled)
             {
+                CleanupTimeoutTask(task);
```

Comment:
maybe this should be in `finally`?

Reply:
In the successful case a few lines below, task result is returned to another handler, which then does some processing and disposes the message.

Reply:
yeah I thought about some condition in the finally